jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/ResponseHttpFields.java

```diff
@@ -173,18 +197,34 @@ public void remove()
             {
                 if (_committed.get())
                     throw new UnsupportedOperationException("Read Only");
+                if (isPersistent(_current))
+                    throw new IllegalStateException("Persistent field");
                 i.remove();
+                _current = null;
```

Comment:
You should also throw ISE when `_current` is null according to the `Iterator.remove()` javadoc.

Reply:
I believe the i.remove() call above will do that.  I can explicitly do the check if you really want ?